pkg/actuators/machine/machine.go

```diff
@@ -251,12 +424,34 @@ func (ms *machineScope) getClusterAddress(ctx context.Context) (map[string]int,
 	return clusterAddr, nil
 }
 
-func (ms *machineScope) reconcileMachineProviderStatus(status *string, id *string) error {
+func (ms *machineScope) findUsableInternalAddress(ctx context.Context, vmID string, excludeAddr map[string]int) (string, error) {
+
+	nicRegex := regexp.MustCompile(`^(eth|en|br\-ex).*`)
+	IPParams := ovirtC.NewVMIPSearchParams().WithIncludedInterfacePattern(nicRegex)
+	nics, err := ms.ovirtClient.GetVMIPAddresses(vmID, IPParams)
```

Comment:
Please use the context on all API calls.

Reply:
done